Source/BugsnagMetadata.m

```diff
@@ -86,15 +81,22 @@ - (void)addAttribute:(NSString *)attributeName
         if (value) {
             id cleanedValue = BSGSanitizeObject(value);
             if (cleanedValue) {
-                [self getTab:tabName][attributeName] = cleanedValue;
-            } else {
+                NSDictionary *section = [self getMetadata:tabName];
+                if (!section) {
+                    [[self dictionary] setObject:[NSMutableDictionary new] forKey:tabName];
+                    section = [self getMetadata:tabName];
```

Comment:
Would it make sense to hold onto the `[NSMutableDictionary new]` here rather than call into `[self getMetadata]`? It's been a while since I looked at Objective-C in detail but I believe that would avoid checking a lock via the `@synchronized` keyword. Are there any potential downsides that mean using `[self getMetadata]` would be preferable?

Reply:
No downsides I can see.  Good spot.